cmd/thanos/compact.go

```diff
@@ -184,6 +185,11 @@ func runCompact(
 		httpserver.WithTLSConfig(conf.http.tlsConfig),
 	)
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Why do you need to create ctx here? Why not just do it at the original line?

Reply:
I had moved it there to be closer to the `tracer` code. Moved it back now. Thanks!